resources/packs/aws/aws_ssm.go

```diff
@@ -0,0 +1,149 @@
+}
+
+func (s *mqlAwsSsm) GetInstances() ([]interface{}, error) {
+	provider, err := awsProvider(s.MotorRuntime.Motor.Provider)
+	if err != nil {
+		return nil, err
+	}
+	res := []interface{}{}
+	poolOfJobs := jobpool.CreatePool(s.getInstances(provider), 5)
+	poolOfJobs.Run()
+
+	// check for errors
+	if poolOfJobs.HasErrors() {
+		return nil, poolOfJobs.GetErrors()
+	}
+	// get all the results
+	for i := range poolOfJobs.Jobs {
+		res = append(res, poolOfJobs.Jobs[i].Result.([]interface{})...)
+	}
+
+	return res, nil
+}
+
+func (s *mqlAwsSsm) getInstances(provider *aws_provider.Provider) []*jobpool.Job {
+	tasks := make([]*jobpool.Job, 0)
+	var err error
+	account, err := provider.Account()
+	if err != nil {
+		return []*jobpool.Job{{Err: err}} // return the error
+	}
+	regions, err := provider.GetRegions()
+	if err != nil {
+		return []*jobpool.Job{{Err: err}} // return the error
+	}
+	log.Debug().Msgf("regions being called for ssm instance list are: %v", regions)
+	for ri := range regions {
+		region := regions[ri]
+		f := func() (jobpool.JobResult, error) {
+			res := []interface{}{}
+			ssmsvc := provider.Ssm(region)
+			ctx := context.Background()
+
+			input := &ssm.DescribeInstanceInformationInput{
+				Filters: []types.InstanceInformationStringFilter{},
+			}
+			nextToken := aws.String("no_token_to_start_with")
+			ssminstances := make([]types.InstanceInformation, 0)
+			for nextToken != nil {
+				isssmresp, err := ssmsvc.DescribeInstanceInformation(ctx, input)
+				if err != nil {
+					return nil, errors.Wrap(err, "could not gather ssm information")
+				}
+				nextToken = isssmresp.NextToken
+				if isssmresp.NextToken != nil {
+					input.NextToken = nextToken
+				}
+				ssminstances = append(ssminstances, isssmresp.InstanceInformationList...)
+			}
+
+			log.Debug().Str("account", account.ID).Str("region", region).Int("instance count", len(ssminstances)).Msg("found ec2 ssm instances")
+			for _, instance := range ssminstances {
+				mqlInstance, err := s.MotorRuntime.CreateResource("aws.ssm.instance",
+					"instanceId", core.ToString(instance.InstanceId),
+					"pingStatus", string(instance.PingStatus),
+					"ipAddress", core.ToString(instance.IPAddress),
+					"platformName", core.ToString(instance.PlatformName),
+					"region", region,
+					"arn", ssmInstanceArn(account.ID, region, core.ToString(instance.InstanceId)),
+				)
+				if err != nil {
+					return nil, err
+				}
+				res = append(res, mqlInstance)
+			}
+			return jobpool.JobResult(res), nil
+		}
+		tasks = append(tasks, jobpool.NewJob(f))
+	}
+	return tasks
+}
+
+func ssmInstanceArn(account string, region string, id string) string {
+	return fmt.Sprintf(ssmInstanceArnPattern, region, account, id)
+}
+
+func (s *mqlAwsSsmInstance) id() (string, error) {
+	return s.Arn()
+}
+
+const ssmInstanceArnPattern = "arn:aws:ssm:%s:%s:instance/%s"
+
+func (d *mqlAwsSsmInstance) init(args *resources.Args) (*resources.Args, AwsSsmInstance, error) {
+	if len(*args) > 2 {
```

Comment:
Same here as for the EC2 instance:
`> 1` or adding a name to the args. I'm not sure whether these instances always have a name...

Reply:
same here, it could be initialized with id and region instead. the name changes easily, so the name isn't really valuable here